src/webgl/p5.RendererGL.js

```diff
@@ -1110,58 +841,63 @@ p5.RendererGL = class RendererGL extends Renderer {
       tmp.draw(() => {
         this.states.filterShader.setUniform('direction', [1, 0]);
         this.states.filterShader.setUniform('tex0', target);
-        this._pInst.clear();
-        this._pInst.shader(this.states.filterShader);
-        this._pInst.noLights();
-        this._pInst.plane(target.width, target.height);
+        this.clear();
+        this.shader(this.states.filterShader);
+        this.noLights();
+        this.plane(target.width, target.height);
       });
 
       // Vert pass: draw `tmp` to `fbo`
       fbo.draw(() => {
         this.states.filterShader.setUniform('direction', [0, 1]);
         this.states.filterShader.setUniform('tex0', tmp);
-        this._pInst.clear();
-        this._pInst.shader(this.states.filterShader);
-        this._pInst.noLights();
-        this._pInst.plane(target.width, target.height);
+        this.clear();
+        this.shader(this.states.filterShader);
+        this.noLights();
+        this.plane(target.width, target.height);
       });
 
-      this._pInst.pop();
+      this.pop();
     }
     // every other non-blur shader uses single pass
     else {
       fbo.draw(() => {
-        this._pInst.noStroke();
-        this._pInst.blendMode(constants.BLEND);
-        this._pInst.shader(this.states.filterShader);
+        this.states.doStroke = false;
+        this.blendMode(constants.BLEND);
+        this.shader(this.states.filterShader);
         this.states.filterShader.setUniform('tex0', target);
         this.states.filterShader.setUniform('texelSize', texelSize);
         this.states.filterShader.setUniform('canvasSize', [target.width, target.height]);
         // filterParameter uniform only used for POSTERIZE, and THRESHOLD
         // but shouldn't hurt to always set
         this.states.filterShader.setUniform('filterParameter', filterParameter);
-        this._pInst.noLights();
-        this._pInst.plane(target.width, target.height);
+        this.noLights();
+        this.plane(target.width, target.height);
       });
 
     }
     // draw fbo contents onto main renderer.
-    this._pInst.push();
-    this._pInst.noStroke();
+    this.push();
+    this.states.doStroke = false;
     this.clear();
-    this._pInst.push();
-    this._pInst.imageMode(constants.CORNER);
-    this._pInst.blendMode(constants.BLEND);
+    this.push();
+    this.states.imageMode = constants.CORNER;
+    this.blendMode(constants.BLEND);
     target.filterCamera._resize();
-    this._pInst.setCamera(target.filterCamera);
-    this._pInst.resetMatrix();
+    this.setCamera(target.filterCamera);
+    this.resetMatrix();
     this._drawingFilter = true;
-    this._pInst.image(fbo, -target.width / 2, -target.height / 2,
-      target.width, target.height);
+    this.image(
+      fbo,
+      0, 0,
+      this.width, this.height,
+      -target.width / 2, -target.height / 2,
+      target.width, target.height
+    );
```

Comment:
@davepagurek Perhaps I got this signature wrong?

Reply:
Thanks, I had a look but still couldn't figure out where the problem is. It might be in how the filter is applied.

Reply:
I'll take a look tonight and see if I can figure out what's going on!